src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs

```diff
@@ -244,16 +244,49 @@ private static OdbcParameter[] DeriveParametersFromStoredProcedure(OdbcConnectio
                 }
             }
             retcode = hstmt.CloseCursor();
-            return rParams.ToArray(); ;
+            return rParams.ToArray();
         }
 
         public override string QuoteIdentifier(string unquotedIdentifier)
         {
             return QuoteIdentifier(unquotedIdentifier, null /* use DataAdapter.SelectCommand.Connection if available */);
         }
+
         public string QuoteIdentifier(string unquotedIdentifier, OdbcConnection connection)
         {
-            throw ADP.NotSupported();
+            ADP.CheckArgumentNull(unquotedIdentifier, nameof(unquotedIdentifier));
+
+            // if the user has specificed a prefix use the user specified  prefix and suffix
+            // otherwise get them from the provider
+            string quotePrefix = QuotePrefix;
+            string quoteSuffix = QuoteSuffix;
+            if (string.IsNullOrEmpty(quotePrefix))
+            {
+                if (connection == null)
+                {
+                    // Use the adapter's connection if QuoteIdentifier was called from 
+                    // DbCommandBuilder instance (which does not have an overload that gets connection object)
+                    connection = DataAdapter?.SelectCommand?.Connection;
```

Comment:
Looks like this line in the reference source is "connection = base.GetConnection() as OdbcConnection; "

Do we not have GetConnection() in Core?